packages/apollo-server-core/src/utils/dispatcher.ts

```diff
@@ -1,19 +1,34 @@
-import { GraphQLRequestListener } from "apollo-server-plugin-base";
+import { AnyFunction, AnyFunctionMap } from "apollo-server-types";
 
-type AnyFunction = (...args: any[]) => any;
 type Args<F> = F extends (...args: infer A) => any ? A : never;
 type AsFunction<F> = F extends AnyFunction ? F : never;
 type UnwrapPromise<T> = T extends Promise<infer U> ? U : T;
 
 type DidEndHook<TArgs extends any[]> = (...args: TArgs) => void;
 
-export class Dispatcher<T extends GraphQLRequestListener> {
+export class Dispatcher<T extends AnyFunctionMap> {
   constructor(protected targets: T[]) {}
 
+  private callTargets<TMethodName extends keyof T>(
+    targets: T[],
+    methodName: TMethodName,
+    ...args: Args<T[TMethodName]>
+  ) {
+    return targets.map(target => {
+      const method = target[methodName];
+      if (method && typeof method === 'function') {
+        return method.apply(target, args);
+      }
+    });
+  }
+
   public async invokeHookAsync<TMethodName extends keyof T>(
     methodName: TMethodName,
     ...args: Args<T[TMethodName]>
   ): Promise<UnwrapPromise<ReturnType<AsFunction<T[TMethodName]>>>[]> {
+    return await Promise.all(
+      this.callTargets(this.targets, methodName, ...args),
+    );
     return await Promise.all(
```

Comment:
You forgot to remove this.

Reply:
Resolved via: b7ea44792af147a9fad6349537753ea9184788eb